### PR TITLE
Added artical list

### PR DESCRIPTION
**Pull request by FeLL1kS**
image

### CODE REVIEW

File: src/styles.scss [hunk 1]
```diff
@@ -0,0 +1,176 @@
+$enable-flex:               true;
```
Comment (anton62k) [inline src/styles.scss:1]: Can we use `demo.productionready.io/main.css` styles or at least https://angular.io/guide/component-styles approach?

File: src/app/article.service.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { Article, MockArticles } from './aricles-mock';
+
+@Injectable({
```
Comment (anton62k) [inline src/app/article.service.ts]: I would suggest a separate "articles" module.
Then we can add `ArticleService` to providers in a module declaration and remove `providedIn: 'root'`.

File: src/app/article.service.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  providedIn: 'root',
+})
+export class ArticleService {
+  constructor() {}
+
+  getArticles(): Observable<Article[]> {
```
Comment (anton62k) [inline src/app/article.service.ts]: There are two problems.

1. Every time we call `getArticles` we return a new instance of this stream.
2. We cannot "emit" updates for articles.

File: src/app/articles-list-item/articles-list-item.component.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Component, Input } from '@angular/core';
+import { Article } from '../aricles-mock';
+
+@Component({
+  selector: 'app-articles-list-item',
+  templateUrl: './articles-list-item.component.html',
+  styleUrls: ['./articles-list-item.component.scss'],
+})
+export class ArticlesListItemComponent {
+  @Input() article?: Article;
```
Comment (anton62k) [inline src/app/articles-list-item/articles-list-item.component.ts]: De facto it is not an optional property. FOr this real-world application, I think we can use something like this

```
@Input() article!: Article;
```

In this case, we can avoid the checks like `if (this.article) {`

File: src/app/articles-list-item/articles-list-item.component.html [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<div class="post-preview" *ngIf="article">
```
Comment (anton62k) [inline src/app/articles-list-item/articles-list-item.component.html]: Angular renders the template when @Input article is not undefined.

File: src/app/articles-list/articles-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  articles: Article[] = [];
+
+  constructor(private articleService: ArticleService) {}
+
+  ngOnInit(): void {
+    this.getArticles();
+  }
+
+  getArticles(): void {
+    this.articleService
```
Comment (anton62k) [inline src/app/articles-list/articles-list.component.ts]: We have to unsubscribe from this stream when the component will be unmounted.

File: src/app/article/article.component.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+  ngOnInit(): void {
+    this.getArticle();
+  }
+
+  getArticle(): void {
+    const slug: string | null = this.route.snapshot.paramMap.get('slug');
+
+    if (slug) {
+      this.articleService
```
Comment (anton62k) [inline src/app/article/article.component.ts]: The same, we have to unsubscribe from the stream.

File: src/app/article/article.component.html [hunk 8]
```diff
@@ -0,0 +1,97 @@
+<div class="post-page" *ngIf="article">
```
Comment (anton62k) [inline src/app/article/article.component.html:1]: The angular way would be using async pipe

```suggestion
<div class="post-page" *ngIf="(article$ | async) as article">
```

```
export class ArticleComponent implements OnInit {
  article$: Obserbable<Article>;
```

File: src/app/article.service.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    const articles = of(MockArticles);
+    return articles;
+  }
+
+  getArticleBySlug(slug: string): Observable<Article | undefined> {
```
Comment (anton62k) [inline src/app/article.service.ts]: The same

File: src/styles.scss (hunk 1 above)
Comment (FeLL1kS) [reply src/styles.scss:1]: We already use this main.css file by reference.  And as for this code, I got it from https://github.com/gothinkster/conduit-bootstrap-template/tree/master/scss repository

**Pull request merged**